src/prefect/cli/deploy.py

```diff
@@ -114,8 +119,8 @@ async def deploy(
             " --work-queue flag."
         ),
     ),
-    work_pool_name: str = typer.Option(
-        None,
+    work_pool_name: str = SettingsOption(
```

Comment:
for my own understanding, what is the benefit of using a `SettingsOption` vs. calling `.value()`?  Does the help documentation display the setting name?

Reply:
Yeah, using setting option will default to the setting value if it isn't called and it will update the help text to show where the default value comes from.

For example from `prefect deploy --help`:
```
--pool      TEXT     The work pool that will handle this deployment's runs. [default: (from PREFECT_DEFAULT_WORK_POOL_NAME)] 
```